userInformation_program.c

```diff
@@ -0,0 +1,208 @@
+    
+
+    int count=1;
+    while (fread(&s,sizeof(s),1,file))
+    {   
+        printf("\t\tUser-%d\n",count++);
+        printf("ID=%d\nName=%s\nAge=%d\n\n",s.ID,s.name,s.age);
+    }
+
+}
+
+void deleteUser(FILE *file){
+    int idToDelete;
+    printf("Enter user's ID you want to delete: ");
+    scanf("%d", &idToDelete);
+
+    FILE *temp = fopen("temp.txt", "w");
+    struct MyStructure s;
+
+    // setting file pointer to start
+    fseek(file, 0, SEEK_SET);
+    
+
+    bool flag=false;
+
+    while (fread(&s, sizeof(s), 1, file)) {
+        if (s.ID == idToDelete) {
+            flag = true;
+            //don't add this  
+        } else {
+            fwrite(&s, sizeof(s), 1, temp);
+        }
+    }
+
+    fclose(temp);
+    fclose(file);
+
+    if (flag==true) {
+        remove("users.txt"); 
+        rename("temp.txt", "users.txt");
+        printf("user deleted\n");
+    } else {
+        remove("temp.txt");
+        printf("\n user not found. \n");
+    }
+
+}
+
+void update(FILE *file){
+    int idToUpdate;
+    printf("Enter user's ID you want to update: ");
+    scanf("%d", &idToUpdate);
+    FILE *temp = fopen("temp.txt", "w");
+    struct MyStructure s;
+
+    // setting file pointer to start
+    fseek(file, 0, SEEK_SET);
+    
+
+    bool flag=false;
+
+   
+
+    while (fread(&s, sizeof(s), 1, file)) {
+        if (s.ID == idToUpdate) {
+
+            char newName[100];
+            int newAge;
+
+            int option;
+            printf("\nWhat you want to update:\n");
+            printf("Enter 1 for name\n");
+            printf("Enter 2 for age\n");
+            printf("Enter 3 for both name and age\n");
+            scanf("%d",&option);
+            switch(option){
+                case 1:
+                    printf("\nEnter new Name:");
+                    scanf("%s",&newName);
+
+                    strcpy(s.name,newName);
+                    break;
+                case 2:
+                    printf("\nEnter new Age:");
+                    scanf("%d",&newAge);
+
+                    s.age=newAge;
+                    break;
+                case 3:
+                    printf("\nEnter new Name:");
+                    scanf("%s",&newName);
+                    printf("\nEnter new Age:");
+                    scanf("%d",&newAge);
+
+                    strcpy(s.name,newName);
+                    s.age=newAge;
+
+                    break;
+                default:
+                    printf("Invalid. Choose from above options only.\n");
+                    break;
+
+            }
+
+
+            flag = true;
+              
+        } 
+            fwrite(&s, sizeof(s), 1, temp);
+        
+    }
+
+    
+    fclose(temp);
+    fclose(file);
+    
+    if (flag) {
+        remove("users.txt"); 
+        rename("temp.txt", "users.txt");
+        printf("user updated\n");
+    } else {
+        remove("temp.txt");
+        printf("\n user not found. \n");
+    }
+
+
+}
+
+int main(){
```

Comment:
In main() function, there should be only function calling.
There should not be things like "taking inputs", etc.